### ISSUE TITLE
Is it possible to execute multiple UPDATES at once?

### ISSUE THREAD

**Issue by fafhrd91** [diesel-rs/diesel#1517]
I couldn't find any information on possibility to execute multiple updates.

could anyone point me to a api doc or example?

**Comment by sgrif** [comment]
Can you expand on what you mean by "execute multiple updates"? Can you provide the SQL query you'd like to construct?

**Comment by fafhrd91** [comment]
something like:

```
UPDATE posts SET field='1' WHERE ..;
UPDATE posts SET field='2' WHERE ..;
UPDATE posts SET field='2' WHERE ..;
```

**Comment by sgrif** [comment]
You just need to call `execute` on multiple update statements. e.g.

```rust
update(posts).set(field.eq("1")).filter(...).execute(&conn)?;
update(posts).set(field.eq("2")).filter(...).execute(&conn)?;
```

**Comment by fafhrd91** [comment]
would it execute each update separately?

some background, I use diesel for emtech framework benchmarks and it performs very well.
but result for "db update" benchmark is very weak. I see other frameworks do batch updates

https://github.com/TechEmpower/FrameworkBenchmarks/tree/master/frameworks/Rust/actix

**Comment by sgrif** [comment]
I haven't seen much real world usage that needs to execute multiple unrelated update statements in a single round trip. If you want to optimize for that benchmark, you can pass a SQL string to `connection.batch_execute` (this is how migrations are run).

**Comment by fafhrd91** [comment]
Thanks.

**Comment by thomasetter** [comment]
Another approach which also works with prepared statements is to chain the updates with `WITH` to create a single statement containing several updates:
https://dba.stackexchange.com/questions/171123/use-same-postgres-with-in-multiple-queries

**Comment by mjanda** [comment]
It seems like it's using `update ... from` syntax and building that update sql manually.

https://github.com/TechEmpower/FrameworkBenchmarks/blob/master/frameworks/Rust/actix/src/db_pg.rs#L180

Nice, didn't even know postgresql can do that.

**Comment by denniscollective** [comment]
First off thanks for the library!

I'm hitting this issue trying to batch upsert. I want to sync against a remote source of truth, but have local records related to that.

I tried playing around with making a Vec of Changesets, but hit a bunch of dead ends before googling and finding this.

Would a pull request be welcomed?

**Comment by ThePianoDentist** [comment]
@dnsco If you dont mind extra-lines/boilerplate and you are using postgresql.

This works for me

```
    diesel::insert_into(table)
        .values(new)
        .on_conflict(league_id)
        .do_update()
        .set((
            meta.eq(excluded(meta)),
            name.eq(excluded(name)),
            team_size.eq(excluded(team_size)),
            competition_id.eq(excluded(competition_id)),
            squad_size.eq(excluded(squad_size)),
            max_players_per_team.eq(excluded(max_players_per_team)),
            max_players_per_position.eq(excluded(max_players_per_position)),
        ))
        //.set(new) would work for single upsert, but not bulk upsert. As the Vec doesnt impl AsChangeset
        .get_results(conn)
```

Extra notes:
I was interested in 'partial-bulk-upsert', where null values indicate "Dont update. leave as-is".
However this looks very tricky/effort.

because if you do it in raw SQL, you'll notice it tries the INSERT with the null values (likely to error for un-nullable colums), before anything else....so it errors with "cant insert null", rather than hitting the on conflict for Pkey.

Whilst could write some postgresql hooks/functions to achieve this, is awkward and far outside realm of diesel.

**Comment by evanxg852000** [comment]
Not sure if this was the original intent. but I hit this link searching for a related issue.
This solved my issue
```
let users_to_deactiave = vec![2,3,4];
diesel::update(splits::table.filter(users::id.eq_any(users_to_deactiave)))
                .set(users::is_active.eq(false))
                .execute(&*conn)?;
```

**Comment by Fuuzetsu** [comment]
Could we re-open? This still seems very desirable.

Something like in https://stackoverflow.com/questions/18797608/update-multiple-rows-in-same-query-using-postgresql

The issue with the solution like https://github.com/diesel-rs/diesel/issues/1517#issuecomment-818000075 is that it _only_ works if you want to perform a very simple update to some field(s). It doesn't work if you have say 20k records and you need to some column values to something different for each one.

Basically if there was a way to pass a slice/vector to `update` + `set` and have that turned into some `UPDATE ... FROM`

**Comment by weiznich** [comment]
@Fuuzetsu Commenting on old issues is not the right way to request new features. Please open a new [thread here](https://github.com/diesel-rs/diesel/discussions/categories/ideas) and address at least the following points:
* What should exactly be added to diesel?
* How would the corresponding public API look?
* Which backends support this SQL variant. 
* Is this supported for all versions of a backend or only for specific versions 
* Do other backends support the same feature with different syntax?
*Are there any invariants, which should be checked at compile time?

**Comment by Fuuzetsu** [comment]
> @Fuuzetsu Commenting on old issues is not the right way to request new features. Please open a new [thread here](https://github.com/diesel-rs/diesel/discussions/categories/ideas) and address at least the following points:
> 

Thanks, will do.

**Comment by Fuuzetsu** [comment]
Continued at https://github.com/diesel-rs/diesel/discussions/2879

**Comment by weiznich** [comment]
@MostHated That's definitively the wrong place to for your issue, as this is not even relevant to this specific issue at all. If you feel that this is an issue with diesel please fill out a new bug report providing all relevant information. Additionally: If that only happens with rust-analyzer and not with rustc that indicates this is not a diesel issue but a rust-analyzer issue as the compiler is considered to be the source of truth. 
I mark your comment as off-topic for the reasons outlined above.
